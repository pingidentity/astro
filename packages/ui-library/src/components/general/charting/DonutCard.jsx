import React, { Component } from "react";
import PropTypes from "prop-types";
import { PieChart, Pie, Cell } from "recharts";
import DashboardCard from "./Cards/DashboardCard";
import FormDropDownList from "../../forms/FormDropDownList";
import classnames from "classnames";

/**
 * @typedef {Object} DonutCard~data
 * @param {string} [id]
 *     Identifier for this data label
 * @param {string|number} [value]
 *     value of the data
 * @param {string} [color]
 *     color of the data
 *
 *
 * @class DonutCard
 * @desc A card that displays a donut chart on a dashboard card.
 *
 * @param {string} [className]
 *    Custom class name(s) applied to the top-level container
 * @param {array} [data-id="donut-card"]
 *  The data-id attribute applied to the top-level container
 * @param {string} [label]
 *     label for the card
 * @param {string} [title]
 *     Title of the card. Displayed at top of front and back
 * @param {number} [value]
 *     The valuee shown on the front and back of the card
 * @param {string} [errorMessage]
 *    When provided, the error message and icon will display in place of the regular front/back content.
 * @param {boolean} [loading=false]
 *    When true the splinner animation shows in place of the stats
 * @param {function} [onMouseOver]
 *     Callback triggered when a chart data-point is hovered over
 * @param {function} [onMouseOut]
 *     Callback triggered when the mouse is hovered out of the chart
 * @param {object} [options]
 *     provides the list of objects for the drop down
 * @param {object} [selectedOption]
 *     highlightes selected option
 * @param {object} [onSelect]
 *     selects from the list of options
 * @param {function} [onMakeDefault]
 *    Callback triggered when the make-default checkbox is changed.  When provided a checkbox will render on the back
 *    of the card.
 * @param {object} [Array.DonutCard~data]
 *     A list of objects that provides the data for the chart
 *  @param {bool} [defaultChecked]
 *     state of the checkbox set to false. If set to true will render with onMakeDefault already checked.
 * */




class DonutCard extends Component {

    state = {
        statColor: null,
        strokeColor: null,
        strokeWidth: 1,
        hoveredSection: null,
    }


    _mouseOver = /* istanbul ignore next  */ (value, index, e) => {

        if (this.props.onMouseOver) {
            this.props.onMouseOver(e, value);
        }

        const dataItem = this.props.data[index];
        const color = dataItem.hasOwnProperty("color") ? dataItem.color : null;

        this.setState({
            hoveredSection: value.id,
            statColor: color,
            strokeColor: color,
            strokeWidth: 2,
        });
    };

    _mouseOut = /* istanbul ignore next  */ (value, index, e) => {
        if (this.props.onMouseOut) {
            this.props.onMouseOut(e, value);
        }

        this.setState({
            hoveredSection: null,
            statColor: null,
            strokeWidth: 1,
        });
    };

    _renderCells = (data) => {
        return data.map(({ id, color }) => {
            return this.state.hoveredSection === id ? (
                <Cell
                    className="donut-card__hovered" key={id}
                    fill={color}
                    style={{
                        stroke: this.state.strokeColor,
                        strokeWidth: this.state.strokeWidth,
                    }}
                />) : (<Cell key={id} fill={color}/>);
        });
    };

    _renderNumber = (value) => {
        if (value >= 1000000) {
            return `${Number.parseFloat((value / 1000000).toFixed(2))}m`;
        } else if (value >= 100000 ) {
            return `${Number.parseFloat((value / 1000).toFixed(0))}k`;
        } else if (value >= 10000) {
            return `${Number.parseFloat((value / 1000).toFixed(1))}k`;
        } else if (value >= 1000) {
            return `${Number.parseFloat((value / 1000).toFixed(2))}k`;
        } else {
            return value;
        }
    };

    _renderCommas = (value) => {
        return value.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ",");
    };

    _renderBack = (data) => {
        return (
            <div className="dashboard-card__stat-list" data-id={`${this.props["data-id"]}-list`} key="dashKey">
                {data.map(({ id, label, value, color }) => (
                    <div key={id} className="donut-card__back-info">
                        <div className="dashboard-card__stat-row-label">{label || id}</div>
                        <div
                            className="dashboard-card__stat-row-number"
                            style={{ color: color }}>
                                {this._renderCommas(value)}
                        </div>
                    </div>
                    ))}
            </div>
        );
    };

    render () {

        const classes = classnames(
            "donut-card",
            this.props.className,
        );

        return (
            <DashboardCard {...this.props} data-id={this.props["data-id"]} className={classes}
                front={(
                    <div>
                        <div className="donut-card__title">
                            {this.props.title}
                        </div>
                        {!this.props.loading && ([
                            <PieChart
                                key="chartKey"
                                height={202}
                                width={202}
                                data-id={`${this.props["data-id"]}-chart`}
                                className="donut-card__donut-chart"
                            >
                                <Pie
                                    innerRadius={60}
                                    outerRadius={100}
                                    paddingAngle={1}
                                    key="pieKey"
                                    data={this.props.data}
                                    dataKey="value"
                                    nameKey="id"
                                    onMouseOver={this._mouseOver}
                                    onMouseOut={this._mouseOut}
                                >
                                    {this._renderCells(this.props.data)}
                                </Pie>
                            </PieChart>,
                            <div key="centerLabelKey" className="donut-card__center-info">
                                <div className="donut-card__center-number" style={{ color: this.state.statColor }}>
                                    {this._renderNumber(this.props.value)}
                                </div>
                                <div className="donut-card__front-line" />
                                <div className="donut-card__center-label" style={{ color: this.state.statColor }}>
                                    {this.props.label}
                                </div>
                            </div>
                        ])}
                    </div>
                )}
                back={(
                    <div>
                        <div
                            data-id={`${this.props["data-id"]}-back-title`}
                            className="dashboard-card__back-title donut-card__back-title"
                            >
                            {this.props.title}
                        </div>
                        {!this.props.loading && ([
                            <div key="backLineKey" className="donut-card__back-line" />,
                            <FormDropDownList
                                data-id={`${this.props["data-id"]}-drop-down`}
                                key="dropdown"
                                className="donut-card__drop-down"
                                options={this.props.options}
                                selectedOption={this.props.selectOption}
                                onValueChange={this.props.onSelect}
                            />,
                            this._renderBack(this.props.data)
                        ])}
                    </div>
                )}
            />
        );
    }




    static propTypes = {
        "data-id": PropTypes.string,
        label: PropTypes.string,
        title: PropTypes.string,
        className: PropTypes.string,
        value: PropTypes.number,
        onMouseOver: PropTypes.func,
        onMouseOut: PropTypes.func,
        innerRadius: PropTypes.number,
        outerRadius: PropTypes.number,
        options: PropTypes.arrayOf(PropTypes.object),
        selectOption: PropTypes.object,
        onSelect: PropTypes.func,
        onMakeDefault: PropTypes.func,
        errorMessage: PropTypes.string,
        data: PropTypes.arrayOf(PropTypes.shape({
            label: PropTypes.string,
            value: PropTypes.number,
            color: PropTypes.string,
        })).isRequired,
    };

    static defaultProps = {
        "data-id": "donut-card",
    };
}

export default DonutCard;