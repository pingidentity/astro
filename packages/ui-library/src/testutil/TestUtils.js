/*eslint-disable valid-jsdoc*/

"use strict";

/**
 * @module util/TestUtils
 * @desc A collection of ReactJS test utils, to complement the ones provided by the
 * framework. This is a copy of the TestUtils in the library, just because this class
 * is used in the tests and should work with the same version of React used in the WP
 * and not in the library.
 */

var ReactDOM = require("react-dom");
var React = require("react");
var ReactTestUtils = require("react-addons-test-utils");
var _ = require("underscore");
var assign = require("object-assign");

var TestUtils = {

    /**
     * @alias module:util/TestUtils.scryRenderedDOMNodesWithDataId
     * @desc Return all DOM nodes children of the given tree with the specified dataId.
     *
     * @param {object} parent - the React component or DOM Node tree to search
     * @param {string} dataId - the value of the data-id attribute on the target elements
     * @return {array} - of DOM node objects which match the given criteria
     */
    scryRenderedDOMNodesWithDataId: function (parent, dataId) {
        // first check if the parent component is a DOM node
        if (ReactDOM.findDOMNode(parent) === null) {
            return [];
        }

        var parentNode =
            ReactTestUtils.isDOMComponent(parent) || ReactTestUtils.isCompositeComponent(parent)
                ? ReactDOM.findDOMNode(parent)
                : parent;

        var nodes;
        if (parentNode.getAttribute("data-id") === dataId) {
            nodes = [parentNode];
        }
        else {
            var selectorValue = dataId
                .replace(/\[/g, "\\[")
                .replace(/\]/g, "\\]")
                .replace(/'/g, "\\'");
            nodes = parentNode.querySelectorAll("[data-id=" + selectorValue + "]");
        }

        return nodes;
    },

    /**
     * @alias module:util/TestUtils.findRenderedDOMNodeWithDataId
     * @desc Return the single DOM node child of the given tree with the specified dataId.
     * If no node in the tree matches the criteria, null is returned.
     * If there are more than one node, an error is thrown.
     *
     * @param {object} parent - the React component or DOM Node tree to search
     * @param {string} dataId - the value of the data-id attribute on the target element
     * @return {object} - the DOM node object which matches the given criteria
     */
    findRenderedDOMNodeWithDataId: function (parent, dataId) {
        var nodes = this.scryRenderedDOMNodesWithDataId(parent, dataId);

        if (nodes.length > 1) {
            throw new Error("Found more than one DOM node with dataId: " +
                dataId + " in the given tree");
        }

        return nodes.length === 0 ? null : nodes[0];
    },

    /**
     * @alias module:util/TestUtils.scryRenderedDOMNodesWithTag
     * @desc Return all DOM nodes children of the given tree with the specified tag name.
     *
     * @param {object} parent - the React component or DOM Node tree to search
     * @param {string} tagName - the tag name on the target elements
     * @return {array} - array of DOM nodes object which match the given criteria
     */
    scryRenderedDOMNodesWithTag: function (parent, tagName) {
        // first check if the parent component is a DOM node
        if (ReactDOM.findDOMNode(parent) === null) {
            return [];
        }

        var parentNode =
            ReactTestUtils.isDOMComponent(parent) || ReactTestUtils.isCompositeComponent(parent)
                ? ReactDOM.findDOMNode(parent)
                : parent;

        var nodes;
        if (parentNode.tagName.toLowerCase() === tagName.toLowerCase()) {
            nodes = [parentNode];
        }
        else {
            nodes = parentNode.querySelectorAll(tagName);
        }

        return nodes;
    },

    /**
     * @alias module:util/TestUtils.findRenderedDOMNodeWithTag
     * @desc Return the single DOM node child of the given tree with the specified tag name.
     * If no node in the tree matches the criteria, null is returned.
     * If there are more than one node, an error is thrown.
     *
     * @param {ReactComponent} parent - the React component or DOM Node tree to search
     * @param {string} tagName - the tag name on the target element
     * @return {DOMNode} - the DOM node which matches the given criteria
     */
    findRenderedDOMNodeWithTag: function (parent, tagName) {
        var nodes = this.scryRenderedDOMNodesWithTag(parent, tagName);

        if (nodes.length > 1) {
            throw new Error("Found more than one DOM node with tag name: " +
                tagName + " in the given tree");
        }

        return nodes.length === 0 ? null : nodes[0];
    },

    /**
     * @alias module:util/TestUtils.scryRenderedDOMNodesWithName
     * @desc Return all DOM nodes children of the given tree with the specified name.
     *
     * @param {ReactComponent} parent - the React component or DOM Node tree to search
     * @param {string} name - the name on the target elements
     * @return {DOMNode[]} - all DOM nodes which match the given criteria
     */
    scryRenderedDOMNodesWithName: function (parent, name) {
        // first check if the parent component is a DOM node
        if (ReactDOM.findDOMNode(parent) === null) {
            return [];
        }

        var parentNode =
            ReactTestUtils.isDOMComponent(parent) || ReactTestUtils.isCompositeComponent(parent)
                ? ReactDOM.findDOMNode(parent)
                : parent;

        var nodes;
        if (parentNode.name === name) {
            nodes = [parentNode];
        }
        else {
            nodes = parentNode.querySelectorAll("[name='" + name.replace("'", "\\'") + "']");
        }

        return nodes;
    },

    /**
     * @alias module:util/TestUtils.findRenderedDOMNodeWithName
     * @desc Return the single DOM node child of the given tree with the specified name.
     * If no node in the tree matches the criteria, null is returned.
     * If there are more than one node, an error is thrown.
     *
     * @param {ReactComponent} parent - the React component or DOM Node tree to search
     * @param {string} name - the name on the target element
     * @return {DOMNode} - the DOM node which matches the given criteria
     */
    findRenderedDOMNodeWithName: function (parent, name) {
        var nodes = this.scryRenderedDOMNodesWithName(parent, name);

        if (nodes.length > 1) {
            throw new Error("Found more than one DOM node with name: " +
                name + " in the given tree");
        }

        return nodes.length === 0 ? null : nodes[0];
    },

    /**
     * @alias module:util/TestUtils.scryRenderedDOMNodesWithClass
     * @desc Return all DOM nodes children of the given tree with the specified CSS class name.
     *
     * @param {ReactComponent} parent - the React component or DOM Node tree to search
     * @param {string} className - the CSS class name on the target elements (or a concatenation of classes, separated by space)
     * @return {DOMNode[]} - all DOM nodes which match the given criteria
     */
    scryRenderedDOMNodesWithClass: function (parent, className) {
        // first check if the parent component is a DOM node
        if (ReactDOM.findDOMNode(parent) === null) {
            return [];
        }

        var parentNode = ReactTestUtils.isDOMComponent(parent) || ReactTestUtils.isCompositeComponent(parent)
            ? ReactDOM.findDOMNode(parent)
            : parent;

        var nodes;

        // first check if the root element has the provided classes;
        // if not, build a selector and look for any children

        var parentMatches = _.every(className.split(" "), function (name) {
            return (" " + parentNode.className + " ").indexOf(" " + name + " ") > -1;
        });
        if (parentMatches) {
            nodes = [parentNode];
        }
        else {
            var selector = _.reduce(className.split(" "), function (memo, name) {
                return memo + "." + name;
            }, "");
            nodes = parentNode.querySelectorAll(selector);
        }

        return nodes;
    },

    /**
     * @alias module:util/TestUtils.findRenderedDOMNodeWithClass
     * @desc Return the single DOM node child of the given tree with the specified CSS class name.
     * If no node in the tree matches the criteria, null is returned.
     * If there are more than one node, an error is thrown.
     *
     * @param {ReactComponent} parent - the React component or DOM Node tree to search
     * @param {string} className - the CSS class name on the target element
     * @return {DOMNode} - the DOM node which matches the given criteria
     */
    findRenderedDOMNodeWithClass: function (parent, className) {
        var nodes = this.scryRenderedDOMNodesWithClass(parent, className);

        if (nodes.length > 1) {
            throw new Error("Found more than one DOM node with class name: " +
                className + " in the given tree");
        }

        return nodes.length === 0 ? null : nodes[0];
    },

    /**
     * @alias module:util/TestUtils.scryRenderedComponentsWithType
     * @desc Finds all instances of components with type equal to `componentType`.
     * @param {object} root - a composite (ie. React) component
     * @param {object} componentType - the component type
     * @return {array} an array of all the matches.
     */
    scryRenderedComponentsWithType: function (root, componentType) {
        return ReactTestUtils.scryRenderedComponentsWithType(root, componentType);
    },

    /**
     * @alias module:util/TestUtils.findRenderedComponentWithType
     * @desc Same as `scryRenderedComponentsWithType` but expects there to be one result
     * and returns that one result, or throws exception if there is any other
     * number of matches besides one.
     * @param {object} root - a composite (ie. React) component
     * @param {object} componentType - the component type
     * @return {!ReactComponent} The one match.
     */
    findRenderedComponentWithType: function (root, componentType) {
        return ReactTestUtils.findRenderedComponentWithType(root, componentType);
    },

    /**
     * @alias module:util/TestUtils.captureGlobalListener
     * @desc Attaches proxy function for `window.addEvenListener` which will save reference to actual provided callback when
     * event type match filter. Useful for testing ReactJS component global (window) callbacks behaviour.
     *
     * @param {string} eventType event to save callback for. case insensitive.
     * @param {object} [obj=window] optional object to capture listener from. Like `document`. If not provided - window object
     * @returns {function} - when executed will trigger actual provided callback to `window.addEventListener`
     *
     * @example
     * //capture registered callback from reactjs control
     * var onKeyDown = TestUtils.captureGlobalListener("keyDown");
     *
     * onKeyDown({keyCode:13});  //later in test trigger global event
     *
     * @example
     * //capture registered callback from reactjs control on `document` object
     * var onKeyDown = TestUtils.captureGlobalListener("keyDown", document);
     *
     * onKeyDown({keyCode:13});  //later in test trigger global event
     */
    captureGlobalListener: function (eventType, obj) {

        var captured;

        var target = obj || window;

        target.addEventListener = function (event, listener) {
            if (event.toLowerCase() === eventType.toLowerCase()) {
                captured = listener;
            }
        };

        return function () {
            return captured.apply(this, arguments);
        };
    }

    /**
     * @desc Return all components in the supplied tree with the specified dataId.
     *
     * @param {ReactComponent} tree - the React component tree to search
     * @param {string} dataId - the value of the data-id attribute on the target elements
     * @return {ReactComponent[]} - all components which match the given criteria
     */
    /*
     scryRenderedDOMComponentsWithDataId: function (tree, dataId) {
     var components = ReactTestUtils.findAllInRenderedTree(tree, function (inst) {
     var instId = ReactDOM.findDOMNode(inst).getAttribute("data-id");
     return ReactTestUtils.isDOMComponent(inst) && (instId === dataId);
     });

     return components;
     },
     */

    /**
     * @desc Return the single component in the supplied tree with the specified dataId.
     * If no component in the tree matches the criteria, null is returned.
     * If there are more than one component, an error is thrown.
     *
     * @param {ReactComponent} tree - the React component tree to search
     * @param {string} dataId - the value of the data-id attribute on the target element
     * @return {ReactComponent} - the component which match the given criteria
     * @throws Will throw an Error if more than one element match the criteria
     */
    /*
     findRenderedDOMComponentWithDataId: function (tree, dataId) {
     var components = TestUtils.scryRenderedDOMNodesWithDataId(tree, dataId);

     if (components.length > 1) {
     throw new Error("Found more than one component with dataId: " +
     dataId + " in the given tree");
     }

     return components.length === 0 ? null : components[0];
     },
     */

    /**
     * @desc Return the single component in the supplied tree with the specified name.
     * If no component in the tree matches the criteria, null is returned.
     * If there are more than one component, an error is thrown.
     *
     * @param {ReactComponent} tree - the React component tree to search
     * @param {string} name - the value of the name attribute on the target element
     * @return {ReactComponent} - the component which match the given criteria
     * @throws Will throw an Error if more than one element match the criteria
     */
    /*
     findRenderedDOMComponentWithName: function (tree, name) {
     var components = ReactTestUtils.findAllInRenderedTree(tree, function (inst) {
     var instName = ReactDOM.findDOMNode(inst).name;
     return ReactTestUtils.isDOMComponent(inst) && (name === instName);
     });

     if (components.length > 1) {
     throw new Error("Found more than one component with name: " +
     name + " in the given tree");
     }

     return components.length === 0 ? null : components[0];
     }
     */
};

/**
 * @class UpdatePropsWrapper
 *
 * @desc
 * A simple wrapper to be used for unit testing, when you need to update
 * a property on a rendered component. Starting with React 0.14.x,
 * directly setting or replacing a prop on a rendered component is deprecated.
 * Instead, use this wrapper and call _setProps(newProps) on it
 * to have the wrapped component re-render with the new property(ies).
 * The ref attribute on the wrapper is set to "wrapper".
 *
 * @Example
 *  // initial render
 *  var component = ReactTestUtils.renderIntoDocument(
 *      <UpdatePropsWrapper type={If} test={true} />
 *  );
 *  // update the property through re-render
 *  component._setProps({test: false});
 *
 * @Example
 *  var component = ReactTestUtils.renderIntoDocument(
 *      <UpdatePropsWrapper type={TabbedSections}
 *              onSectionChange={jest.genMockFunction()}
 *              selectedIndex={-1}>
 *          <div data-id="section1" title="section 1">section 1</div>
 *          <div data-id="section2" title="section 2">section 2</div>
 *      </UpdatePropsWrapper>
 *  );
 *  component._setProps({ selectedIndex: 0 });
 *
 */
TestUtils.UpdatePropsWrapper = React.createClass({
    /**
     * Re-render the wrapped component using the new property.
     * Use this function to avoid setting the props directly on the component (anti pattern).
     * @param {object} props - new property(ies) to set
     */
    _setProps: function (props) {
        this.setState(props);
    },
    getInitialState: function () {
        return assign({ ref: "wrapper" }, this.props);
    },
    render: function () {
        return React.createElement(
            this.props.type, // this is the type of element to create, provided as prop
            this.state // this is the rest of props to set on the element
        );
    }
});

module.exports = TestUtils;
