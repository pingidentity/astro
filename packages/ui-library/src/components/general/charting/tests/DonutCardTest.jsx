import React from "react";
import ReactTestUtils from "react-dom/test-utils";
import TestUtils from "../../../../testutil/TestUtils";
import _ from "underscore";
import DonutCard from "../DonutCard";


describe("DonutCard", function () {

    const componentId = "donut-card";

    function getComponent (opts) {
        const withDefaults = _.defaults(opts || {}, {
            "data-id": componentId,
            data: [
                { id: "Enabled Users", value: 120543 , color: "#E12F51" },
                { id: "Inactive Users", value: 51233, color: "green" },
                { id: "Disabled Users", value: 3000, color: "pink" },
            ],
            options: [
                { label: "Current", value: "1" },
                { label: "30 DAYS", value: "2" },
                { label: "60 DAYS", value: "3" },
                { label: "90 DAYS", value: "4" },
            ]
        });

        return ReactTestUtils.renderIntoDocument(<DonutCard {...withDefaults} />);
    }

    it("rendered component with data-id=donut-card", function () {
        const component = getComponent({
        });

        const element = TestUtils.findRenderedDOMNodeWithDataId(component, componentId);

        expect(ReactTestUtils.isDOMComponent(element)).toBeTruthy();
    });

    it("renders a title", function () {
        const component = getComponent({
            title: "title"
        });

        const element = TestUtils.findRenderedDOMNodeWithDataId(component, componentId);

        expect(element).toBeTruthy();
    });

    it("renders a label and number", function () {
        const component = getComponent({
            label: "title",
            value: 3
        });

        const element = TestUtils.findRenderedDOMNodeWithDataId(component, componentId);

        expect(element).toBeTruthy();
    });

    it("renders a number greater than 1000", function () {
        const component = getComponent({
            value: 10000
        });

        const element = TestUtils.findRenderedDOMNodeWithDataId(component, componentId);

        expect(element).toBeTruthy();
    });

    it("renders a dropdown", function() {
        const component = getComponent({});
        const dropDown = TestUtils.findRenderedDOMNodeWithDataId(component, "donut-card-drop-down");

        expect(dropDown).toBeDefined();
    });

    it("renders a Pie Chart", function() {
        const component = getComponent({});
        const pieChart = TestUtils.findRenderedDOMNodeWithDataId(component, "donut-card-chart");

        expect(pieChart).toBeDefined();
    });

    it("renders a back title", function() {
        const component = getComponent({});
        const title = TestUtils.findRenderedDOMNodeWithDataId(component, "donut-card-back-title");

        expect(title).toBeDefined();
    });


    it("renders hr tag", function() {
        const component = getComponent({});

        const element = TestUtils.scryRenderedDOMNodesWithTag(component, "hr");

        expect(element.length).toEqual(2);
    });

    it("renders an svg", function() {
        const component = getComponent({});

        const element = TestUtils.scryRenderedDOMNodesWithTag(component, "svg");

        expect(element.length).toEqual(3);
    });

    it("should render the state color", function() {
        const component = getComponent({});
        component.setState({ statColor: "red" });

        expect(component).toBeTruthy();
    });

    it("should render the state color to null", function() {
        const component = getComponent({});
        component.setState({ statColor: null });

        expect(component).toBeTruthy();
    });

});