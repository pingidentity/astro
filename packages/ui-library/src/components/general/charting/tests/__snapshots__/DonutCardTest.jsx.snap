// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DonutCard data-id's don't change 1`] = `
Array [
  "donut-card",
  "donut-card-drop-down",
  "selected-option",
  "selected-input",
  "text-field_container",
  "selected-input-input",
  "donut-card-list",
  "padding-component",
  "pie-chart",
  "donut-card-view-toggle",
]
`;
